content/TruenasApps/UsingCustomApp.md

```diff
@@ -48,138 +35,1199 @@ If you are unsure about any configuration settings, review the [Install Custom A
+      DB_PORT: "5432"
+      DB_USERNAME: immich
+      IMMICH_LOG_LEVEL: log
+      IMMICH_MACHINE_LEARNING_ENABLED: "true"
+      IMMICH_MACHINE_LEARNING_URL: http://machine-learning:32002
+      IMMICH_PORT: "8080"
+      NODE_ENV: production
+      NVIDIA_DRIVER_CAPABILITIES: all
+      NVIDIA_VISIBLE_DEVICES: GPU-UUID # Replace me
+      REDIS_DBINDEX: "0"
+      REDIS_HOSTNAME: redis
+      REDIS_PASSWORD: immich
+      REDIS_PORT: "6379"
+      TZ: Etc/UTC
+    healthcheck:
+      test:
+        - CMD-SHELL
+        - '/bin/bash -c ''exec {health_check_fd}<>/dev/tcp/127.0.0.1/8080 && echo -e "GET /api/server-info/ping HTTP/1.1\r\nHost: 127.0.0.1\r\nConnection: close\r\n\r\n" >&$${health_check_fd} && cat <&$${health_check_fd}'''
+      timeout: 5s
+      interval: 10s
+      retries: 30
+      start_period: 10s
+    image: altran1502/immich-server:v1.117.0
+    links:
+      - pgvecto
+      - redis
+      - machine-learning
+    networks:
+      default: null
+    ports:
+      - mode: ingress
+        host_ip: 0.0.0.0
+        target: 8080
+        published: "8080"
+        protocol: tcp
+    restart: unless-stopped
+    security_opt:
+      - no-new-privileges
+    volumes:
+      - type: volume
+        source: library
+        target: /usr/src/app/upload/library
+        volume: {}
+      - type: volume
+        source: uploads
+        target: /usr/src/app/upload/upload
+        volume: {}
+      - type: volume
+        source: thumbs
+        target: /usr/src/app/upload/thumbs
+        volume: {}
+      - type: volume
+        source: profile
+        target: /usr/src/app/upload/profile
+        volume: {}
+      - type: volume
+        source: video
+        target: /usr/src/app/upload/encoded-video
+        volume: {}
+      - type: volume
+        target: /tmp
+        volume: {}
+      - type: volume
+        target: /scratchpad
+        volume:
+          nocopy: true
+networks:
+  default:
+    name: ad7a9839cdf6cde9e34b526956ba5f37_default
+volumes:
+  library:
+    name: ad7a9839cdf6cde9e34b526956ba5f37_library
+  postgres-data:
+    name: ad7a9839cdf6cde9e34b526956ba5f37_postgres-data
+  profile:
+    name: ad7a9839cdf6cde9e34b526956ba5f37_profile
+  thumbs:
+    name: ad7a9839cdf6cde9e34b526956ba5f37_thumbs
+  tmp:
+    name: ad7a9839cdf6cde9e34b526956ba5f37_tmp
+  uploads:
+    name: ad7a9839cdf6cde9e34b526956ba5f37_uploads
+  video:
+    name: ad7a9839cdf6cde9e34b526956ba5f37_video{{< /codeblock >}}
+
+#### Installing a Multi-Container App with a Dockerfile
+This example deploys Nextcloud with multiple containers, including redis and postgres.
+
+A [Dockerfile](https://docs.docker.com/reference/dockerfile/) builds the Nextcloud image locally, rather than pulling from a remote repository.
+You can use an existing Dockerfile or, as in the example, use [`dockerfile_inline`](https://docs.docker.com/reference/compose-file/build/#dockerfile_inline) to define the build commands as an inline string within the Compose file.
```

Comment:
I don't know that we need to capitalize dockerfile and compose in these sentences as we are not taking about proper nouns. Thoughts?
```suggestion
A [dockerfile](https://docs.docker.com/reference/dockerfile/) builds the Nextcloud image locally, rather than pulling from a remote repository.
You can use an existing dockerfile or, as in the example, use [`dockerfile_inline`](https://docs.docker.com/reference/compose-file/build/#dockerfile_inline) to define the build commands as an inline string within the compose file.
```

Reply:
They are proper nouns though. Dockerfile is the name of a specific thing. Compose is being used as the name of the [Docker] Compose software, not "compose," which is a verb.

A Google search shows most other online docs (Docker or otherwise) are capitalizing Dockerfile